core/src/main/java/org/fao/geonet/kernel/security/openidconnect/SimpleOidcUser.java

```diff
@@ -0,0 +1,228 @@
+import org.springframework.security.oauth2.core.oidc.OidcIdToken;
+import org.springframework.security.oauth2.core.oidc.StandardClaimNames;
+import org.springframework.util.StringUtils;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Contains information about the user (from the oidc ID Token).
+ * <p>
+ * This class knows how to do two things;
+ * a) extra info from the OIDC ID Token
+ * b) move info to a GN user
+ * <p>
+ * parts taken from BaselUser (GN keycloak plugin)
+ */
+class SimpleOidcUser {
+
+
+    private String username;
+    private String firstname;
+    private String surname;
+    private String organisation;
+    private String profile;
+    private String email;
+    private Address address;
+
+
+    SimpleOidcUser(OIDCConfiguration oidcConfiguration, OIDCRoleProcessor oidcRoleProcessor, OidcIdToken idToken) {
+        username = idToken.getPreferredUsername();
+        if (username == null) {
+            username = idToken.getFullName();
+        }
+        if (username != null) {
+            username = org.apache.commons.lang.StringUtils.left(username, 256); //first max 256 chars
+        }
+
+        if (username != null && username.length() > 0) {
+            surname = idToken.getFamilyName();
+            firstname = idToken.getGivenName();
+            email = idToken.getEmail();
+
+
+            if (idToken.getClaims() != null && idToken.getClaims().containsKey(oidcConfiguration.organizationProperty)) {
+                organisation = (String) idToken.getClaims().get(oidcConfiguration.organizationProperty);
+            }
+
+
+            if (idToken.getAddress() != null) {
+                address = new Address();
+                address.setAddress(idToken.getAddress().getStreetAddress());
+                address.setCity(idToken.getAddress().getLocality());
+                address.setState(idToken.getAddress().getRegion());
+                address.setZip(idToken.getAddress().getPostalCode());
+                address.setCountry(idToken.getAddress().getCountry());
+            }
+
+
+            Map<Profile, List<String>> profileGroups = oidcRoleProcessor.getProfileGroups(idToken);
+            if (profileGroups != null && profileGroups.size() > 0) {
+                profile = oidcRoleProcessor.getMaxProfile(profileGroups).name();
+            }
+        }
+    }
+
+    SimpleOidcUser(OIDCConfiguration oidcConfiguration, OIDCRoleProcessor oidcRoleProcessor, Map attributes) {
+        username = (String) attributes.get(oidcConfiguration.getUserNameAttribute());
+        if (username == null)
+            username = (String) attributes.get(StandardClaimNames.PREFERRED_USERNAME);
+        if (username == null) {
+            username = (String) attributes.get(StandardClaimNames.NAME);
+        }
+        if (username != null) {
+            username = org.apache.commons.lang.StringUtils.left(username, 256); //first max 256 chars
+        }
+
+        if (username != null && username.length() > 0) {
```

Comment:
Change to `!org.apache.commons.lang.StringUtils.isBlank(username)`?

Reply:
changed